src/assets/wise5/directives/componentAnnotations/component-annotations.component.ts

```diff
@@ -45,34 +57,88 @@ export class ComponentAnnotationsComponent {
         }
       }
     );
+    this.wiseLinkCommunicatorId = `wise-link-communicator-component-annotations-${this.nodeId}-${this.componentId}`;
   }
 
-  ngOnChanges() {
+  ngAfterViewInit() {
     this.processAnnotations();
+    this.wiseLinkCommunicator = document.getElementById(this.wiseLinkCommunicatorId);
+    if (this.wiseLinkCommunicator != null) {
```

Comment:
Is this null check necessary?

Reply:
I added nodeId and componentId as input parameters to component-annotations. Currently only multiple-choice-student passes those two parameters in. All the other components don't pass in nodeId and componentId in yet which is why we need the null check. We could either change all components to pass in nodeId and componentId or keep the null check until we eventually do pass in nodeId and componentId after we've converted all the components to Angular.

Reply:
I see. It looks like this.wiseLinkCommunicator will never be null even if nodeId and componentId are not passed in though. And it seems to work without any problems when I remove the null checks in ngAfterViewInit() and ngOnDestroy().

this.wiseLinkCommunicatorId becomes "wise-link-communicator-component-annotations-undefined-undefined" and this.wiseLinkCommunicator is a div element.

So we should either:
1. Use the undefined id (remove null-checks)
2. Check for nodeId/componentId null
2. Pass in nodeId and componentId to all the component-annotations. 

Since things are already working without the null-check, probably option 1 or 3?